docs/stencil-docs/reference-docs/stencil-utils-api-reference.md

```diff
@@ -52,13 +51,34 @@ updateView() {
     this.refreshView(content);
   });
 }
- ```
+```
+
+The Cornerstone theme uses the `getPage` method in the following contexts:
 
-### Usage in Cornerstone
 - [`assets/js/theme/wishlist.js`](https://github.com/bigcommerce/cornerstone/blob/master/assets/js/theme/wishlist.js)
 - [`assets/js/theme/gift-certificate.js`](https://github.com/bigcommerce/cornerstone/blob/master/assets/js/theme/gift-certificate.js)
 - [`assets/js/theme/faceted-search.js`](https://github.com/bigcommerce/cornerstone/blob/master/assets/js/theme/common/faceted-search.js)
 
+### Get page by GraphQL
+
+[getPageByGQL](https://github.com/bigcommerce/stencil-utils/blob/master/src/api/index.js)
+
+This method takes the form `utils.api.getPageByGQL(page, callback)`. 
+
+It allows you to render parts of a template using an inline GraphQL query, as shown in [Front Matter Reference - GraphQL attributes](/stencil-docs/reference-docs/front-matter-reference#graphql-attributes).
+
+| Parameter | Type | Description/Usage |
+|:----------|:-----|:------------------|
+| page | string | Presentation template; for example, `pages/store-locator`. <br> The `template` option allows you to select a particular template, or an array of templates, to render one page. Each value must correspond to a file in the theme's `templates/pages/` subdirectory. |
+| callback | function | Asynchronous function call to handle the results. |
+
+```js title="Example: getPageByGQL" lineNumbers
+utils.api.getPageByGQL('pages/store-locator', (err, response) => {
+  if (err) return;
+  modal.updateContent(response);
```

Comment:
please indicate datatype of response. you can use code comments.

unless there's a reason not to, throw or return the error in the the example. if errors should return undefined (what this does now), please note that in code comments.

Reply:
@bc-andreadao 
Actually, we can miss the handling of the first argument (error) since the server returns the HTML template anyway. it doesn't matter if it's an error or a successful response. The response always is returned as a second argument. This case `if (err) return;` is more likely for the unexpected to catch the error case.


<img width="845" alt="Screenshot 2023-01-23 at 11 27 40" src="https://user-images.githubusercontent.com/84462142/214005701-dc6972cb-6f38-493d-8380-6f6f5fa1c6ba.png">
<img width="877" alt="Screenshot 2023-01-23 at 11 27 48" src="https://user-images.githubusercontent.com/84462142/214005707-9fdd7100-2e73-4904-8b76-2dce2ef18772.png">

Reply:
@bc-apostoliuk Thanks for the info. Do you have the datatype of the response?

Reply:
@bc-andreadao  this is `HTML` template.

Reply:
@bc-andreadao @bc-apostoliuk the idea here is to demonstrate handling the unexpected result, so I'm going to flesh this out a little.